### PR TITLE
Minor code cleanup + tests update + remove CLI arguments and make config file mandatory

### PR DESCRIPTION
**Pull request by Geod24**
See each commit. Let me know if you prefer it split up across multiple PRs.

### CODE REVIEW

File: docs/config.example.yaml [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
-  # Port on which we bind
-  port:    3836    # default value
+  # Address to which we bind (default: 127.0.0.1)
+  address: "0.0.0.0"
+  # Port on which we bind (default: 3836)
+  port:    4242
 
   # Endpoint of Agora
-  agora_endpoint: http://127.0.0.1:2826
+  agora_endpoint: http://8.8.8.8:4567
```
Comment (TrustHenry) [inline docs/config.example.yaml]: dns.google ?
I wonder why you set `8.8.8.8` as an example.

Comment (Geod24) [reply docs/config.example.yaml]: It's not the default, just an invalid example. I could also use `127.1.1.1`

Comment (Geod24) [reply docs/config.example.yaml]: Fixed

File: tests/Config.test.ts [hunk 2]
```diff
@@ -44,30 +44,15 @@ describe('Test of Config', () => {
...
-        let config: Config = new Config(null);
-        config.readFromFile(path.resolve(Utils.getInitCWD(), "tests/config.example.yaml"));
-        assert.strictEqual(config.server.address, "");
-        assert.strictEqual(config.server.port.toString(), "3836");
-        assert.strictEqual(config.server.agora_endpoint.toString(), "http://127.0.0.1:2826");
+        let config: Config = new Config();
+        config.readFromFile(path.resolve(process.cwd(), "docs/config.example.yaml"));
+        assert.strictEqual(config.server.address, "0.0.0.0");
+        assert.strictEqual(config.server.port.toString(), "4242");
+        assert.strictEqual(config.server.agora_endpoint.toString(), "http://8.8.8.8:4567");
```
Comment (TrustHenry) [inline tests/Config.test.ts]: It is strange that this syntax is being tested.

Comment (Geod24) [reply tests/Config.test.ts]: Fixed 🙈

**Review by TrustHenry — APPROVED** [review]
LGTM!

**Pull request merged**